File: src/main/java/org/cubexmc/storage/StorageProvider.java
package org.cubexmc.storage;

import org.bukkit.configuration.file.FileConfiguration;

/**
 * Storage boundary for mutable RuleGems runtime data.
 */
public interface StorageProvider {

    String getName();

    void initialize();

    FileConfiguration readGemData();

    void saveGemData(FileConfiguration data);
}

File: src/main/java/org/cubexmc/storage/StorageProvider.kt
package org.cubexmc.storage

import org.bukkit.configuration.file.FileConfiguration

/**
 * Storage boundary for mutable RuleGems runtime data.
 */
interface StorageProvider {
    fun getName(): String

    fun initialize()

    fun readGemData(): StorageLoadResult

    fun saveGemData(data: FileConfiguration): StorageSaveResult
}
